### ISSUE TITLE
Can DPOTrainer support inputting encoded token IDs

### ISSUE THREAD

**Issue by LBJ6666** [huggingface/trl#2368]
### Feature request

Can DPOTrainer support inputting encoded token IDs to customize the calculation of different attention masks, such as excluding the prompt part from loss computation? For example:
```
model_inputs['prompt_input_ids'] = prompt_input_ids
model_inputs['prompt_attention_mask'] = [torch.ones_like(input_ids) for input_ids in prompt_input_ids]
model_inputs['chosen_input_ids'] = chosen_input_ids
model_inputs['chosen_attention_mask'] = [torch.ones_like(input_ids) for input_ids in chosen_input_ids]
model_inputs['rejected_input_ids'] = rejected_input_ids
model_inputs['rejected_attention_mask'] = [torch.ones_like(input_ids) for input_ids in rejected_input_ids]
```

Currently it seems to only support text input before encoding 
```
model_inputs['prompt'] = prompt
model_inputs['chosen'] = chosen
model_inputs['rejected'] = rejected
```

### Motivation

excluding the prompt part from loss computation
or
customizing the length of intercepted prompt and message inputs

### Your contribution

none

**Comment by qgallouedec** [comment]
> Can DPOTrainer support inputting encoded token IDs to customize the calculation of different attention masks

No, and won't be supported, unless we are provided with a good reason to support it.

> excluding the prompt part from loss computation?

Actually, it's how dpo works by default. See

https://github.com/huggingface/trl/blob/b80c1a6fb8754c578f7178213e56d780abbe96d5/trl/trainer/dpo_trainer.py#L1089-L1091

**Comment by LBJ6666** [comment]
@qgallouedec Thank you for your response
